Barotrauma/BarotraumaClient/ClientSource/GUI/GUITextBox.cs

```diff
@@ -838,6 +927,53 @@ public void SelectAll()
             CalculateSelection();
         }
 
+        private static int FindStopSymbolLeft(string text, int idx)
+        {
+            // special case: if cursor at the start of the line, move to the end of previous line
+            if (idx > 0 && text[idx - 1] == '\n')
+            {
+                return idx - 1;
+            }
+            // convert cursor index to character index
+            idx = Math.Max(0, idx - 2);
+            if (idx == 0) return idx;
+
+            for (; idx >= 0; idx--)
+            {
+                var c = text[idx];
+                if (char.IsWhiteSpace(c) || char.IsPunctuation(c) || c == '\n')
+                {
+                    // convert character index back cursor index
+                    return idx + 1;
+                }
+            }
+            return Math.Clamp(idx, 0, text.Length);
+        }
+
+        private static int FindStopSymbolRight(string text, int idx)
+        {
+            // special case: if cursor is at the end of line, move to the start of next line
+            if (idx < text.Length && text[idx] == '\n')
```

Comment:
Accessing the string array without first checking whether the index is within the valid range will throw an index out of range exception with "incorrect" or unexpected parameters, like `FindStopSymbolRight("Test", 100)`.